src/Orleans.DurableJobs/DurableJobRunResult.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+
+namespace Orleans.DurableJobs;
+
+/// <summary>
+/// Represents the result of a durable job execution.
+/// </summary>
+[GenerateSerializer]
+public sealed class DurableJobRunResult
+{
+    private DurableJobRunResult(DurableJobRunStatus status, TimeSpan? pollAfter, Exception? exception)
+    {
+        Status = status;
+        PollAfterDelay = pollAfter;
+        Exception = exception;
+    }
+
+    /// <summary>
+    /// Gets the status of the job execution.
+    /// </summary>
+    [Id(0)]
+    public DurableJobRunStatus Status { get; }
+
+    /// <summary>
+    /// Gets the delay before the next status check when <see cref="Status"/> is <see cref="DurableJobRunStatus.PollAfter"/>.
+    /// </summary>
+    [Id(1)]
+    public TimeSpan? PollAfterDelay { get; }
+
+    /// <summary>
+    /// Gets the exception associated with a failed job execution when <see cref="Status"/> is <see cref="DurableJobRunStatus.Failed"/>.
+    /// </summary>
+    [Id(2)]
+    public Exception? Exception { get; }
+
+    /// <summary>
+    /// Creates a result indicating the job completed successfully.
+    /// </summary>
+    /// <returns>A completed job result.</returns>
+    public static DurableJobRunResult Completed() => new(DurableJobRunStatus.Completed, null, null);
+
+    /// <summary>
+    /// Creates a result indicating the job should be polled again after the specified delay.
+    /// </summary>
+    /// <param name="delay">The time to wait before checking the job status again.</param>
+    /// <returns>A poll-after job result.</returns>
+    /// <remarks>
+    /// The job will remain in an inline polling loop without being re-queued.
+    /// The polling loop will hold a concurrency slot until the job completes or fails.
+    /// TODO: Add validation for minimum/maximum poll delays to prevent abuse.
+    /// TODO: Consider concurrency slot management for long-running polls.
```

Comment:
This is a good point. We could let the job indicate whether the job is:
- Running
  - actively executing
  - blocks a slot
  - poll after short delay
- Waiting
  - not actively executing,
  - doesn't block a slot
  - should be re-queued, potentially on a different shard)

'Waiting' could be (mis-)used to provide periodic execution, like reminders do today, but it would be the same logical 'run', just delayed/rescheduled.

We could call the methods/statuses:
- `InProgress(TimeSpan estimatedCompletion)`
- `Suspended(DateTimeOffset dueTime)`

What do you think? In this PR, we only need the first. We might not need the second method at all since it can already be implemented by a combination of scheduling a new run + completing this run.